graspy/align/base.py

```diff
@@ -0,0 +1,122 @@
+        if X.shape[1] != Y.shape[1]:
+            msg = "Two datasets have different number of components!"
+            raise ValueError(msg)
+        return X, Y
+
+    @abstractmethod
+    def fit(self, X, Y):
+        """
+        Uses the two datasets to learn the matrix Q_ that aligns the first
+        dataset with the second.
+
+        Parameters
+        ----------
+        X : np.ndarray, shape (n, d)
+            First dataset of vectors. These vectors need to have same number of
+            dimensions as ones in Y, but the number of vectors can differ.
+
+        Y : np.ndarray, shape (m, d)
+            Second dataset of vectors. These vectors need to have same number
+            of dimensions as ones in X, but the number of vectors can differ.
+
+        Returns
+        -------
+        self : returns an instance of self
+        """
+        pass
+
+    def transform(self, X):
+        """
+        Transforms the dataset X using the learned matrix Q_. This may be the
+        same as the first dataset as in .fit(), or a new dataset. For example,
+        additional samples from the same dataset.
+
+        Parameters
+        ----------
+        X : np.ndarray, shape (n, d)
+            Dataset of vectors. Needs to have the same number of dimensions as
+            X and Y passed to fit, but can have a different number of entries.
+
+        Returns
+        -------
+        X_prime : np.ndarray, shape (n, d)
+            First dataset of vectors, aligned to second. Equal to X @ self.Q_.
+        """
+        if not isinstance(X, np.ndarray):
+            msg = f"Dataset is a {type(X)}, not an np.ndarray! "
+            raise TypeError(msg)
+        X = check_array(X)
+        if not X.shape[1] == self.Q_.shape[0]:
+            msg = (
+                "Dataset needs to have the same number of dimensions, d, "
+                "as datasets X and Y used in fit. Currently, vectors in "
+                f"the dataset to transform have {X.shape[1]} dimensions, "
+                f"while vectors in fit had {self.Q_.shape[0]} dimensions."
+            )
+            raise ValueError(msg)
+        return X @ self.Q_
+
+    def fit_transform(self, X, Y):
+        """
+        Uses the two datasets to learn the matrix Q_ that aligns the first
+        dataset with the second. Then, transforms the first dataset accordingly
+        and returns it.
+
+        Parameters
+        ----------
+        X : np.ndarray, shape (n, d)
+            First dataset of vectors. These vectors need to have same number of
+            dimensions as ones in Y, but the number of vectors can differ.
+
+        Y : np.ndarray, shape (m, d)
+            Second dataset of vectors. These vectors need to have same number
+            of dimensions as ones in X, but the number of vectors can differ.
+
+        Returns
+        -------
+        X_prime : np.ndarray, shape (n, d)
+            First dataset of vectors, aligned to second. Equal to X @ self.Q_X.
```

Comment:
same comment as throughout about wrapping variables in ticks

Reply:
not sure how one wraps when there are equations involved. this:
```
 ``X`` @ ``self.Q_``
```
?

Reply:
maybe try 
 ``X @ self.Q_``
and see how it looks

also i just realized it looks like you have `Q_X` in there still

Reply:
also, any reason why it is double ticks and not single?

Reply:
> maybe try
> `X @ self.Q_`
> and see how it looks

where do i see docs compiled from a branch?..

> also i just realized it looks like you have `Q_X` in there still

sneaky boi. fxd.

Reply:
i actually dont know about single vs double. single may actually be better after doing some reading, i know double is interpreted as literal and renders like code, maybe single actually lets you do references? not sure. can try if you'd like

Reply:
netlify link is still at the bottom with the other checks

Reply:
> netlify link is still at the bottom with the other checks

i assume i need to add something somewhere in order for the align references to show up here: https://deploy-preview-419--graspy.netlify.app/ ..

Reply:
@bdpedigo to here, i assume https://github.com/microsoft/graspologic/blob/dev/docs/reference/index.rst ?

Reply:
correct! try that and lmk if you have any issues?

Reply:
will also need to make `align.rst` here https://github.com/microsoft/graspologic/tree/dev/docs/reference

Reply:
gotcha

Reply:
these .rst file use different conventions, just fyi (even on dev, it seems)

Reply:
can you elaborate?

Reply:
Yes, so for example:
```
Datasets
========
```
vs. 
```
**********
Clustering
**********
```
Not sure if this one makes practical of difference, or not.

Another example (although this i can believe is a feature, not a bug):
```
.. autofunction:: er_np

.. autofunction:: er_nm

.. autofunction:: sbm

.. autofunction:: rdpg

.. autofunction:: er_corr

.. autofunction:: sbm_corr

.. autofunction:: rdpg_corr
```
vs.
```
Erdos-Reyni models
------------------

.. autoclass:: EREstimator

.. autoclass:: DCEREstimator

Stochastic block models
-----------------------

.. autoclass:: SBMEstimator

.. autoclass:: DCSBMEstimator

Latent position models
----------------------

.. autoclass:: RDPGEstimator
```
( also vs.
```
Heatmap
-------
.. autofunction:: heatmap

Gridplot
--------
.. autofunction:: gridplot

Pairplot
--------
.. autofunction:: pairplot

Degreeplot
----------
.. autofunction:: degreeplot

Edgeplot
--------
.. autofunction:: edgeplot

Screeplot
---------
.. autofunction:: screeplot
```
)
so the simulations are all piled into the same group in the table of contents, estimators are grouped by types and all plotting functions have its own entry (this i can maybe understand if you are making an argument "all simulations are much more similar to each other than all plots, so all simulations go into the same pile, but plots aren't", but is this really intentional?).

i wasn't sure which convention to follow. it seemed like i should be more similar to module like cluster, as opposed to module like simulations, so i made a separate entry for each class.